multiversx_sdk_cli/transactions.py

```diff
@@ -86,14 +90,25 @@ def do_prepare_transaction(args: Any) -> Transaction:
             if relayer_account.address.to_bech32() != tx.relayer:
                 raise IncorrectWalletError("")
 
+            if isinstance(relayer_account, LedgerAccount):
+                tx_computer.apply_options_for_hash_signing(tx)
+
             tx.relayer_signature = bytes.fromhex(relayer_account.sign_transaction(tx))
         except NoWalletProvided:
             logger.warning("Relayer wallet not provided. Transaction will not be signed by relayer.")
         except IncorrectWalletError:
             raise IncorrectWalletError("Relayer wallet does not match the relayer's address set in the transaction.")
 
+    try:
+        guardian_account = get_guardian_account_from_args(args)
+        if isinstance(guardian_account, LedgerAccount):
```

Comment:
Can also extract it to a function such as `_apply_options_for_hash_signing_if_appropriate(account, tx)`, since this logic is applied in three places (a bit of duplication).

Reply:
will do in a future PR.